app/models/user.rb

```diff
@@ -8,7 +8,7 @@ class User < ApplicationRecord
                       format: { with: VALID_EMAIL_REGEX },
                       uniqueness: true
     has_secure_password
-    validates :password, presence: true, length: { minimum: 6 }
+    validates :password, presence: true, length: { minimum: 6 }, allow_nil: true
```

Comment:
Why do you use `allow_nil = true` here? Do you mean still save the user when the password = nil?

Reply:
_has_secure_password_ has already included a validation method that ensures the password can't be empty. moreover, because both the presence validation and the _has_secure_password_ validation catch empty passwords, the signup form used to produce duplicate error messages when the user submits empty passwords. _allow_nil: true_ would fix that problem while allowing the user to leave the password field empty while updating user information.